experimental/dds/tree2/src/feature-libraries/default-field-kinds/optionalField.ts

```diff
@@ -69,30 +163,49 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
 					delete fieldChange.newContent;
 				}
 
-				// The previous changes applied to a different value, so we discard them.
-				// TODO: Represent muted changes
-				newNodeChanges.length = 0;
+				// Node was changed by this revision: flush the current changes
+				if (currentChildNodeChanges.length > 0) {
+					addChildChange(
+						{ revision, localId: change.fieldChange.id },
+						...currentChildNodeChanges,
+					);
+					currentChildNodeChanges = [];
+				}
 
 				if (change.fieldChange.newContent?.changes !== undefined) {
-					newNodeChanges.push(tagChange(change.fieldChange.newContent.changes, revision));
+					currentChildNodeChanges.push(
+						tagChange(change.fieldChange.newContent.changes, revision),
+					);
 				}
 			}
 		}
 
-		const composed: OptionalChangeset = {};
-		if (fieldChange !== undefined) {
-			if (newNodeChanges.length > 0) {
+		if (currentChildNodeChanges.length > 0) {
+			if (fieldChange !== undefined) {
 				assert(
 					fieldChange.newContent !== undefined,
-					0x5c4 /* Shouldn't have new node changes if there is no new node */,
+					"after node must be defined to receive changes",
 				);
-				fieldChange.newContent.changes = composeChild(newNodeChanges);
+				fieldChange.newContent.changes =
+					currentChildNodeChanges.length === 1
+						? currentChildNodeChanges[0].change
+						: composeChild(currentChildNodeChanges);
+			} else {
+				addChildChange("self", ...currentChildNodeChanges);
 			}
+		}
+
+		const composed: OptionalChangeset = {};
+
+		if (fieldChange !== undefined) {
 			composed.fieldChange = fieldChange;
 		}
 
-		if (origNodeChange.length > 0) {
-			composed.childChange = composeChild(origNodeChange);
+		if (perChildChanges.size > 0) {
+			composed.childChanges = Array.from(perChildChanges.entries(), ([id, changeList]) => [
+				id,
+				changeList.length === 1 ? changeList[0].change : composeChild(changeList),
```

Comment:
Same rationale as above.
```suggestion
				composeChild(changeList),
```